### PR TITLE
Fix split torch_function handling

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74727
* #83014
* #74728
* __->__ #83866

`Tensor.split` calls `TensorBase.split` whose `handle_torch_function` statement passes `func` as `Tensor.split` which is usually correct, but not here because of the use of `super()`. Instead this calls `torch._VF.split` which correctly differentiates from `torch.split`. This is currently okay since we never hit `TensorBase.split` for types with `__torch_function__` however, once we  allow skipping only one hop of `__torch_function__` this will expose the error.

### CODE REVIEW

File: torch/_tensor.py [hunk 1]
```diff
@@ -763,16 +763,16 @@ def split(self, split_size, dim=0):
...
+        if isinstance(split_size, Tensor):
             try:
                 split_size = int(split_size)
-                return super(Tensor, self).split(split_size, dim)
             except ValueError:
-                return super(Tensor, self).split_with_sizes(split_size, dim)
+                pass
+
+        if isinstance(split_size, int):
+            return torch._VF.split(self, split_size, dim)  # type: ignore[attr-defined]
```
Comment (albanD) [inline torch/_tensor.py:773]: Why not directly call `torch.split` here?

**Review by albanD — COMMENTED** [review]
refactor sounds good.

File: torch/_tensor.py (hunk 1 above)
Comment (peterbell10) [reply torch/_tensor.py:773]: That wouldn't work because `torch.split` calls `Tensor.split`.

Comment (albanD) [reply torch/_tensor.py:773]: Ho we expose split there from c++ and then we override it 5 lines later. Ok I missed that. Sounds good.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**